### PR TITLE
expose VcpkgRoot so that build script can call exe.addVcpkgPath

### PR DESCRIPTION
**Pull request by Jack-Ji**

### CODE REVIEW

**Comment by fabioarnold** [comment]
Maybe add a test to LibExeObjStep.zig like this so it doesn't break in the future again?

```
test "addVcpkgPaths" {
    var arena = std.heap.ArenaAllocator.init(std.testing.allocator);
    defer arena.deinit();

    var builder = try Builder.create(
        arena.allocator(),
        "test",
        "test",
        "test",
        "test",
    );
    defer builder.destroy();

    var exe = builder.addExecutable("not_an_executable", "/not/an/executable.zig");
    exe.addVcpkgPaths(.dynamic) catch |err| {
        if (err != error.VcpkgNotFound) {
            return err;
        }
    };
}
```

**Comment by Jack-Ji** [comment]
> Maybe add a test to LibExeObjStep.zig like this so it doesn't break in the future again?
> 
> ```
> test "addVcpkgPaths" {
>     var arena = std.heap.ArenaAllocator.init(std.testing.allocator);
>     defer arena.deinit();
> 
>     var builder = try Builder.create(
>         arena.allocator(),
>         "test",
>         "test",
>         "test",
>         "test",
>     );
>     defer builder.destroy();
> 
>     var exe = builder.addExecutable("not_an_executable", "/not/an/executable.zig");
>     exe.addVcpkgPaths(.dynamic) catch |err| {
>         if (err != error.VcpkgNotFound) {
>             return err;
>         }
>     };
> }
> ```

Good idea, it's done [2193fe4](https://github.com/ziglang/zig/pull/13924/commits/2193fe49bf71475c0358e0d6a7c8721e2abb28d3)

File: lib/std/build/LibExeObjStep.zig [hunk 1]
```diff
@@ -2060,3 +2060,24 @@ test "addPackage" {
     const dupe = exe.packages.items[0];
     try std.testing.expectEqualStrings(pkg_top.name, dupe.name);
 }
+
+test "addVcpkgPaths" {
```
Comment (fabioarnold) [inline lib/std/build/LibExeObjStep.zig:2064]:
```suggestion
test "addVcpkgPaths" {
    if (builtin.os.tag == .wasi) return error.SkipZigTest;

```
Sorry, missed the skip for wasi.

**Review by andrewrk — CHANGES_REQUESTED** [review]
I don't think this test is particularly valuable because it's only testing package not found. I'm ok with merging this without a new test, but if we're going to add tests for vcpkg then I think it should be a proper integration test which is out of scope for this PR.

**Comment by Jack-Ji** [comment]
OK, I think I understand your concern. Should I make a new clean PR, or change this one directly? @andrewrk

**Pull request merged**